graphistry/plugins/kustograph.py

```diff
@@ -0,0 +1,82 @@
+import pandas as pd
+import time
+import json
+from typing import Any, Dict, List, Optional
+
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder
+from azure.kusto.data.exceptions import KustoServiceError
+
+from graphistry.Plottable import Plottable
+from graphistry.util import setup_logger
+
+logger = setup_logger(__name__)
+
+
+class KustoConnectionError(Exception):
+    pass
+
+
+class KustoQueryResult:
+    def __init__(self, data: List[List[Any]], column_names: List[str]):
+        self.data = data
+        self.column_names = column_names
+
+
+class KustoGraph:
+
+    def __init__(self, g: Plottable, kusto_config: Dict[str, str]):
+        required_keys = ["cluster", "database"]
+        for key in required_keys:
+            if not kusto_config.get(key):
+                raise ValueError(f"Missing required kusto_config key: '{key}'")
+
+        self.cluster = kusto_config["cluster"]
+        self.database = kusto_config["database"]
+
+        if all(kusto_config.get(k) for k in ["client_id", "client_secret", "tenant_id"]):
+            self.credential_mode = "aad_app"
+            self.client_id = kusto_config["client_id"]
+            self.client_secret = kusto_config["client_secret"]
+            self.tenant_id = kusto_config["tenant_id"]
+        else:
+            self.credential_mode = "device_code"
+
+        self.client = self._connect()
+
+    def _connect(self) -> KustoClient:
```

Comment:
Do we need reconnect / autoreconnect in query fail?

Also a close() ?

Reply:
Out of scope, but lifecycle methods for these may help Louie side

Cc @exrhizo for visibility